packages/flutter_test/lib/src/test_default_binary_messenger.dart

```diff
@@ -302,6 +311,82 @@ class TestDefaultBinaryMessenger extends BinaryMessenger {
     }, handler);
   }
 
+  /// Set a handler for intercepting stream events sent to the
+  /// platform on the given channel.
+  ///
+  /// Intercepted method calls are not forwarded to the platform.
+  ///
+  /// The given handler will replace the currently registered
+  /// handler for that channel, if any. To stop intercepting messages
+  /// at all, pass null as the handler.
+  ///
+  /// Events are decoded using the codec of the channel.
+  ///
+  /// The handler's stream messages are used as a response, after re-encoding
+  /// them using the channel's codec.
+  ///
+  /// To send an error, send a [PlatformException] in the handler.
+  /// Other exceptions are not caught.
+  ///
+  /// {@macro flutter.flutter_test.TestDefaultBinaryMessenger.handlePlatformMessage.asyncHandlers}
+  ///
+  /// Registered handlers are cleared after each test.
+  ///
+  /// See also:
+  ///
+  ///  * [setMockMethodCallHandler], which is the similar method for
+  ///    [MethodChannel].
+  ///
+  ///  * [setMockMessageHandler], which is similar but provides raw
+  ///    access to the underlying bytes.
+  ///
+  ///  * [setMockDecodedMessageHandler], which is similar but decodes
+  ///    the messages using a [MessageCodec].
+  void setMockStreamHandler(EventChannel channel, MockStreamHandler? handler) {
```

Comment:
This new functionality (and the functionality added in mock_event_channel.dart) should be covered with some unit tests in `flutter_test/test` to make sure it continues to work correctly. Regarding my comment above, it would be good to have some tests covering error handling for example.

(I do see you refactored some existing tests in the framework to use this new feature, but there should be dedicated tests in the flutter_test package for this to ensure that we never break this functionality. The test in the framework may go away or get refactored again to use a different mechanism all together.)